sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileTable.scala

```diff
@@ -34,13 +35,21 @@ import org.apache.spark.sql.util.SchemaUtils
 
 abstract class FileTable(
     sparkSession: SparkSession,
-    options: CaseInsensitiveStringMap,
+    originalOptions: CaseInsensitiveStringMap,
     paths: Seq[String],
     userSpecifiedSchema: Option[StructType])
   extends Table with SupportsRead with SupportsWrite {
 
   import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
 
+  // Options without path-related options from `originalOptions`.
+  protected final lazy val options: CaseInsensitiveStringMap = {
+    val caseInsensitiveMap = CaseInsensitiveMap(originalOptions.asCaseSensitiveMap.asScala.toMap)
+    val caseInsensitiveMapWithoutPaths = caseInsensitiveMap - "paths" - "path"
+    new CaseInsensitiveStringMap(
+      caseInsensitiveMapWithoutPaths.asInstanceOf[CaseInsensitiveMap[String]].originalMap.asJava)
+  }
```

Comment:
@cloud-fan Should I move `FileDataSourceV2.getPaths` to this class so that the operations with path-related options are in one place? (Wanted to make sure before making a bigger change).

Reply:
I can't remember why we don't get the `paths` in this parent class. Seems clearer. One thing we need to take care is getting `paths` should be lazy. Please take a try.

also cc @gengliangwang

Reply:
There was a time when the `FileIndex` is created in `FileDataSourceV2`, so that the `getPaths` method was in `FileDataSourceV2`. In the current code, it seems fine to move the location of the method.

Reply:
I tried to move `getPaths` to `FileTable` here: https://github.com/apache/spark/pull/29437/commits/15fd313044649031d4919108509e3827301565b7. That required moving `getName` (which depends on paths) as well, and `FileTable` needs to have a dependency on `DatasourceRegister` to get `shortName`. Also, whether to use the original options or options without paths is still not clear with this approach.

So, I followed @viirya's [comment](https://github.com/apache/spark/pull/29437#discussion_r471933928) where options without paths are passed to `FileTable`, which seems to be more straightforward. The latest commit has this approach.

Please let me know which approach seems better. Thanks!